packages/x-data-grid/src/components/cell/GridBooleanCell.tsx

```diff
@@ -54,6 +54,10 @@ function GridBooleanCellRaw(props: GridBooleanCellProps) {
     [rootProps.slots.booleanCellFalseIcon, rootProps.slots.booleanCellTrueIcon, value],
   );
 
+  if (value === undefined) {
+    return null;
+  }
```

Comment:
Not sure we should do this now, it's a breaking change (currently, `undefined` value is treated as `false`)

Reply:
I added this to avoid this:
<img width="482" alt="image" src="https://github.com/user-attachments/assets/47da8749-ccea-402f-bc0d-c2a6db825c97">

For client-side row grouping we have a concept of autogenerated group nodes and we handle it like this:
https://github.com/mui/mui-x/blob/ab4823da20defea849b98e006dcd35ffec8d65fb/packages/x-data-grid/src/components/cell/GridBooleanCell.tsx#L145-L147

In server-side row grouping, all the rows are regular group rows and nothing is auto generated, so the above check doesn't work. Did a hacky'ish [solution](https://github.com/mui/mui-x/pull/13826/commits/10747b1958906d5d7e88ea72de046bf8656b0ca0), let me know how does it feel.